spring-integration-event/src/main/java/org/springframework/integration/event/inbound/ApplicationEventListeningMessageProducer.java

```diff
@@ -116,20 +129,45 @@ public void onApplicationEvent(ApplicationEvent event) {
 		}
 	}
 
+	@Override
+	protected Object evaluatePayloadExpression(Object rootObject) {
+		if (!this.payloadExpressionSet && rootObject instanceof PayloadApplicationEvent) {
+			return ((PayloadApplicationEvent) rootObject).getPayload();
```

Comment:
The `PayloadApplicationEvent` is something to which the raw object is wrapped when it is published.
See `ApplicationEventPublisher.publishEvent(Object event)` implementation.

So, from my perspective the user expect to see exactly the `object` which has been published. That is a premise of the `@EventListener`. And that's why we are able to map them directly bypassing `ApplicationEvent` custom impl. 
So, if I expect `java.util.Date` I'm really what to have `Date` in the `payload` of the message after this adapter.
See `ApplicationEventListeningMessageProducerTests.testPayloadEvents()`.

Yes, I think I agreed with you that expression evaluation is redundant here. Just because we don't have anything specific with the `PayloadApplicationEvent` to extract. So, we can just ignore it and always send the result of the `getPayload()`.

WDYT now?